platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/DefaultLenientConfiguration.java

```diff
@@ -223,54 +121,23 @@ public Set<ResolvedArtifact> getArtifacts() {
-        DeprecationLogger.deprecateMethod(LenientConfiguration.class, "getArtifacts(Spec)")
-            .withAdvice("Use a lenient ArtifactView with a componentFilter instead.")
-            .willBeRemovedInGradle9()
-            .withUpgradeGuideSection(8, "deprecate_filtered_configuration_file_and_filecollection_methods")
-            .nagUser();
-
-        LenientArtifactCollectingVisitor visitor = new LenientArtifactCollectingVisitor();
-        ResolvedArtifactSet filteredArtifacts = resolveFilteredArtifacts(dependencySpec, getSelectedArtifacts());
-        artifactSetResolver.visitArtifacts(filteredArtifacts, visitor, resolutionHost);
-        resolutionHost.rethrowFailuresAndReportProblems("artifacts", visitor.getFailures());
-        return visitor.artifacts;
-    }
-
-    /**
-     * Returns a subset of {@code artifactResults} accessible from dependencies matching {@code dependencySpec}.
-     */
-    private ResolvedArtifactSet resolveFilteredArtifacts(Spec<? super Dependency> dependencySpec, SelectedArtifactResults artifactResults) {
-        List<ResolvedArtifactSet> artifactSets = new ArrayList<>();
-        for (Map.Entry<FileCollectionDependency, Integer> entry : fileDependencyResults.getFirstLevelFiles().entrySet()) {
-            if (dependencySpec.isSatisfiedBy(entry.getKey())) {
-                artifactSets.add(artifactResults.getArtifactsWithId(entry.getValue()));
-            }
-        }
-
-        CachingDirectedGraphWalker<DependencyGraphNodeResult, ResolvedArtifact> walker = new CachingDirectedGraphWalker<>(new ResolvedDependencyArtifactsGraph(artifactSets));
-        for (DependencyGraphNodeResult node : getFirstLevelNodes(dependencySpec)) {
-            walker.add(node);
-        }
-        walker.findValues();
-        return CompositeResolvedArtifactSet.of(artifactSets);
-    }
-
     private static class LenientArtifactCollectingVisitor implements ArtifactVisitor {
-        final Set<ResolvedArtifact> artifacts = new LinkedHashSet<>();
-        final Set<File> files = new LinkedHashSet<>();
-        List<Throwable> failures;
+
+        private final Set<ResolvedArtifact> artifacts = new LinkedHashSet<>();
+        private @Nullable List<Throwable> failures;
 
         @Override
         public void visitArtifact(DisplayName variantName, ImmutableAttributes variantAttributes, ImmutableCapabilities capabilities, ResolvableArtifact artifact) {
             try {
                 ResolvedArtifact resolvedArtifact = artifact.toPublicView();
-                files.add(resolvedArtifact.getFile());
+
+                // Attempt to download the file
+                resolvedArtifact.getFile();
+
+                // Only record the artifact if the file is accessible
                 artifacts.add(resolvedArtifact);
             } catch (org.gradle.internal.resolve.ArtifactResolveException e) {
-                //ignore
+                // Ignore
```

Comment:
🤔 Pre-existing code, but, should we at least log here?

Reply:
I don't think so. This is used to implement lenient resolution -- this is an expected outcome of running this code.

We should rather not be using exceptions for control flow here

Reply:
Added a TODO that it would be good to not use exceptions as control flow